capa/main.py

```diff
@@ -1244,8 +1250,7 @@ def main(argv=None):
                 log_unsupported_os_error()
                 return E_INVALID_FILE_OS
 
-        meta = collect_metadata(argv, args.sample, args.format, args.os, args.rules, extractor)
-
+        meta = collect_metadata(argv, args.sample, args.format, args.os, args.rules, extractor).to_capa()
         capabilities, counts = find_capabilities(rules, extractor, disable_progress=args.quiet)
         meta["analysis"].update(counts)
         meta["analysis"]["layout"] = compute_layout(rules, extractor, capabilities)
```

Comment:
rather than casting back into the dict representation, lets update the subsequent code to use the `Metadata` type instead. i'd prefer to deprecate the `Metadata.to_capa` API and remove all uses.

Reply:
I was trying to do using the `Metadata` type but i was facing issues while implementing the `update(count)`. Can you suggest any way to do it?

Reply:
can you provide more details on the issue with `update(count)`? what was the problem and what did you try?

Reply:
Hi, @williballenthin @manasghandat  
I think the `update(count)` doesn't directly work on pydantic Basemodel class as MetaData doesn't have any attribute `update`. As `find_capabilities` in main return count type of dict. We can refer to https://stackoverflow.com/questions/73421846/how-to-update-pydantic-model-from-dictionary regarding this.

Reply:
good idea. @Aayush-Goel-04 would you like to continue working on this PR and get it ready for merge?

Reply:
Thanks @williballenthin , Will create a PR soon.

Reply:
> rather than casting back into the dict representation, lets update the subsequent code to use the `Metadata` type instead. i'd prefer to deprecate the `Metadata.to_capa` API and remove all uses.

Will `ResultDocument.to_capa` also be deprecated.